src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2DPaintApplier.java

```diff
@@ -828,6 +841,37 @@ private COSArray buildType2Functions(List<Color> colors, COSArray domain, COSArr
         }
     }
 
+    /**
+     * Get the value of a private field from the specified object.
+     *
+     * @param obj The object containing the field.
+     * @param fieldName The name of the field.
+     * @param <T> The field type.
+     * @return The private field value.
+     */
+    @SuppressWarnings({ "unchecked", "WeakerAccess" })
+    protected static <T> T getPrivateFieldValue(Object obj, String fieldName)
+    {
+        try
+        {
+            Class<?> c = obj.getClass();
+            try
+            {
+                Field f = c.getDeclaredField(fieldName);
+                f.setAccessible(true);
```

Comment:
This wont work in JDK9+ :( There are hacks (Unsafe&Friends) to workaround that, but I am not sure that I want to import them here.

Reply:
Hopefully we can get the patch for PDFBox to make this unnecessary.